src/components/MessageList/LivestreamMessageList.js

```diff
@@ -0,0 +1,502 @@
+
+    scrollDown();
+    // scroll down after images load again
+    setTimeout(scrollDown, 200);
+  };
+
+  /**
+   * target - target to scroll to (DOM element, scrollTop Number, 'top', or 'bottom'
+   * containerEl - DOM element for the container with scrollbars
+   * source: https://stackoverflow.com/a/48429314
+   */
+  scrollToTarget = (target, containerEl) => {
+    // Moved up here for readability:
+    const isElement = target && target.nodeType === 1;
+    const isNumber =
+      Object.prototype.toString.call(target) === '[object Number]';
+
+    let scrollTop;
+    if (isElement) scrollTop = target.offsetTop;
+    else if (isNumber) scrollTop = target;
+    else if (target === 'top') scrollTop = 0;
+    else if (target === 'bottom')
+      scrollTop = containerEl.scrollHeight - containerEl.offsetHeight;
+
+    if (scrollTop !== undefined) containerEl.scrollTop = scrollTop; // eslint-disable-line no-param-reassign
+  };
+
+  goToNewMessages = () => {
+    this.scrollToBottom();
+    this.setState({ newMessagesNotification: false });
+  };
+
+  userScrolledUp = () => this.scrollOffset > 310;
+
+  listenToScroll = (offset) => {
+    this.scrollOffset = offset;
+    if (this.state.newMessagesNotification && !this.userScrolledUp()) {
+      this.setState({ newMessagesNotification: false });
+    }
+  };
+
+  /**
+   * Adds a temporary notification to message list.
+   * Notification will be removed after 5 seconds.
+   *
+   * @param notificationText  Text of notification to be added
+   * @param type              Type of notification. success | error
+   */
+  addNotification = (notificationText, type) => {
+    if (typeof notificationText !== 'string') return;
+    if (type !== 'success' && type !== 'error') return;
+
+    const id = uuidv4();
+
+    this.setState(({ notifications }) => ({
+      notifications: [...notifications, { id, text: notificationText, type }],
+    }));
+
+    // remove the notification after 5000 ms
+    const ct = setTimeout(
+      () =>
+        this.setState(({ notifications }) => ({
+          notifications: notifications.filter((n) => n.id !== id),
+        })),
+      5000,
+    );
+
+    this.notificationTimeouts.push(ct);
+  };
+
+  onMessageLoadCaptured = () => {
+    // A load event (emitted by e.g. an <img>) was captured on a message.
+    // In some cases, the loaded asset is larger than the placeholder, which means we have to scroll down.
+    if (!this.userScrolledUp()) {
+      this.scrollToBottom();
+    }
+  };
+
+  loadMore = () => {
+    if (this.props.messages.length > this.state.slicedMessages.length) {
+      // If channel already has loaded more messages, then simply get more messages from channel (this.props.messages).
+      this.setState({
+        slicedMessages: this.props.messages.slice(
+          -(this.state.slicedMessages.length + this.props.messageLimit),
+        ),
+      });
+      return;
+    }
+
+    // Otherwise fetch from server.
+    if (this.props.messageLimit) {
```

Comment:
Not really familiar with the code, but is this condition necessary? You end up calling `props.loadMore` with an `undefined` value for `messageLimit` anyways

Reply:
passing `undefined` down will replace the `loadMore(limit = 100)` default value, good call though, we can simplify this a bit

Reply:
Makes sense ... have fixed it by adding default value to messageLimit prop and removing this if condition